### PR TITLE
[CODEGEN-1950] Add supported methods to capabilities

### PR DESCRIPTION
**Pull request by baekdahl**
# This repository is public. Do not put here any private DataRobot or customer's data: code, datasets, model artifacts, .etc.

## Summary
DRUM has a `/capabilities` endpoint that can currently be used to determine what kind of payload formats are supported by the model, i.e. CSV, arrow, etc.
This PR will extend this endpoint to include information about supported methods, for now the only method that is included is chat.
The endpoint will be used by the DataRobot app to expose if a deployment supports chat or not.

## Rationale
Make it possible for clients to find out if a custom model deployment supports chat api.

### CODE REVIEW

**Comment by eric-s-s** [comment]
@baekdahl please provide some description in the PR intro

**Comment by baekdahl** [comment]
> @baekdahl please provide a more detailed description in the PR header

@eric-s-s Added some more description.

File: custom_model_runner/datarobot_drum/drum/language_predictors/python_predictor/python_predictor.py [hunk 1]
```diff
@@ -72,7 +72,7 @@ def _dr_api_url(endpoint):
             endpoint = f"{endpoint}/api/v2"
         return endpoint
 
-    def _supports_chat(self):
+    def supports_chat(self):
```
Comment (eric-s-s) [inline custom_model_runner/datarobot_drum/drum/language_predictors/python_predictor/python_predictor.py:75]: is this tested anywhere?  a quick search yielded no tests for this.  

- if it's tested, please provide a link
- if not, please test this check of the custom hooks

Comment (baekdahl) [reply custom_model_runner/datarobot_drum/drum/language_predictors/python_predictor/python_predictor.py:75]: Test `supports_chat` or `has_custom_hook` ?

Comment (baekdahl) [reply custom_model_runner/datarobot_drum/drum/language_predictors/python_predictor/python_predictor.py:75]: Added a test for `supports_chat`

**Review by yakov-g — APPROVED** [review]

**Review by ScottOglesby — APPROVED** [review]

**Pull request merged**